### ISSUE TITLE
Regexp matching needs more documentation / expected match fails

### ISSUE THREAD

**Issue by kontur** [zimme/meteor-iron-router-active#14]
Hey,

I have a URL like this

`http://localhost:3000/venue/4d6e756a922f8cfad1f449fe` 

and try marking it active with

`{{isActiveRoute regex='venue\/[a-z0-9]*$'}}`

This fails, however. I'm unclear on to what the regex actually tries to match against or if it even takes the whole path or only the last bit.

The documentation should either be more explicit on what works, or we should have a look why this example does not trigger. :)

Great package otherwise, just what I needed :+1:

**Comment by zimme** [comment]
`isActiveRoute` matches against the current route name. You'll need to use `isActivePath` to match against the path part of the url. In your case `venue/4d6e756a922f8cfad1f449fe`

But most likely you have named the route that handles this url path to something like `showVenue` and I would match against that with `isActiveRoute` instead.

```html
<li class={{isActiveRoute regex='showVenue'}}>
  {{linkTo route='showVentue'}}
</li>
```

But if you had trouble understanding the readme I'll take a look at it and see if I can't make it more excplicit.

**Comment by kontur** [comment]
I must have just skimmed over the readme too quickly. I can try send you a PR with a more extensive readme documentation of this difference within the next few days. Maybe show one sample router setup with a named route and a matching isActiveRoute expression?

**Comment by zimme** [comment]
:+1:

**Comment by kontur** [comment]
See https://github.com/zimme/meteor-iron-router-active/pull/16 for now.

When testing some routes I was wondering about the regex interpretation here. I tried to match route names with a dot in them and got unexpected results. Some expressions didn't match, and when trying to escape the dot with \. like so `regex='post\.edit'` meteor will crash. I suppose this is a separate issue, but maybe you have an idea what's going on?

**Comment by zimme** [comment]
Regex in javascript is a bit different It seems. I've had this issue but with escaping a `?` and we came to the conclusion that you need to escape with `\\.` or use a character class [.]

I'm looking at #16 now and it's a great start. I'll put some work into my packages tonight... I need to redo some documentation to make things more obvious in all of them.

**Comment by zimme** [comment]
Ok when you try and escape special chars in javascript strings you need to also escape the `\` symbol and that's why you need to use `'a\\.b'` for it to work. However if you use `/a\.b/` format you don't need to escape the `\` char

**Comment by zimme** [comment]
I'll add RegExp literals to the list of possible arguments tonight.

**Comment by kontur** [comment]
:+1: I'll close this original issue then when PR is merged ;)

**Comment by zimme** [comment]
This is being fixed with #22. Closing this issue.
